app/script/assets/AssetMetaDataBuilder.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Wire
+ * Copyright (C) 2016 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.assets = z.assets || {};
+
+// Builder for creating all kinds of asset metadata
+z.assets.AssetMetaDataBuilder = {
+
+  /*
+  Constructs corresponding asset metadata depending on the given file type
+
+  @param {File|Blob} file - the file to generate metadata for
+  @returns {ImageMetaData|VideoMetaData|AudioMetaData}
+  */
+  build_metadata(file) {
+    if (!(file instanceof Blob)) {
+      throw new Error('Expected file to be type of Blob');
+    }
+
+    if (this.is_video(file)) {
+      return this._build_video_metdadata(file);
+    } else if (this.is_audio(file)) {
+      return this._build_audio_metdadata(file);
+    } else if (this.is_image(file)) {
+      return this._build_image_metdadata(file);
+    } else {
+      return Promise.resolve();
+    }
```

Comment:
like I said.. already.. I'm not a big fan of it..! So I won't mind if not changed :)